### DISCUSSION TITLE
Objects and Arrays as Customizable Properties

### DISCUSSION THREAD

**Discussion by EonZeNx** [rocksdanister/lively#691] [Ideas]
I've updated the Fluid sim to have a bunch of extra properties such as colour, location, etc.
I was wondering if it were possible to have an array and object property creator under the Customize menu?

Video for the interested:
[https://v.redd.it/wpjlsv3f8ni71](https://v.redd.it/wpjlsv3f8ni71)

**Comment by rocksdanister** [comment]
>I was wondering if it were possible to have an array and object property creator under the Customize menu?

I don't understand.. how exactly will this look in a gui and can't you just use a dropdown/slider for some pre defined choices instead?

**Reply by EonZeNx** [reply]
I'd like the user to create an arbitrary number of objects in an array, that would bloat the customize menu after only 2 or 3 objects and still limit the user to a few objects.

I was thinking something like the colour picker where you would pop up a new element with the structure of the object each time you click an add new button?

**Reply by rocksdanister** [reply]
You should make a feature request with some mockups and examples if possible:

https://github.com/rocksdanister/lively/issues

**Reply by EonZeNx** [reply]
Created an issue with a mock-up image and some code layout, #693
